apis/XCast/IXCast.h

```diff
@@ -26,51 +26,202 @@
-
-				virtual void onApplicationLaunchRequestWithLaunchParam(const string& appName, const string& strPayLoad, const string& strQuery, const string& strAddDataUrl) = 0;
-				virtual void onApplicationLaunchRequest(const string& appName, const string& parameter) = 0;
-				virtual void onApplicationStopRequest(const string& appName, const string& appID) = 0;
-				virtual void onApplicationHideRequest(const string& appName, const string& appID) = 0;
-				virtual void onApplicationStateRequest(const string& appName, const string& appID) = 0;
-				virtual void onApplicationResumeRequest(const string& appName, const string& appID) = 0;
-				virtual void onUpdatePowerStateRequest(const string& powerState) = 0;
+				// @text onApplicationLaunchRequest
+				// @brief Triggered when the cast service receives a launch request from a client with launch params
+				// @param applicationName: Registered application name
+				// @param strPayLoad: Payload string to be passed to the application
+				// @param strQuery: Query string to be appended in launch request
+				// @param strAddDataUrl: Additional data URL to be passed to the application
+				virtual void OnApplicationLaunchRequestWithParam(const string& appName /* @text applicationName */ , const string& strPayLoad /* @text strPayLoad */, const string& strQuery /* @text strQuery */, const string& strAddDataUrl /* @text strAddDataUrl */) {};
+				// @text onApplicationLaunchRequest
+				// @brief Triggered when the cast service receives a launch request from a client with launch params
+				// @param applicationName: Registered application name
+				// @param parameter: Application launch string
+				virtual void OnApplicationLaunchRequest(const string& appName /* @text applicationName */ , const string& parameter /* @text parameter */ )  {};
+				// @text onApplicationStopRequest
+				// @brief 	Triggered when the cast service receives a stop request from a client
+				// @param applicationName: 	Registered application name
+				// @param applicationId: 	Application instance ID
+				virtual void OnApplicationStopRequest(const string& appName /* @text applicationName */, const string& appID /* @text applicationId */)  {};
+				// @text onApplicationHideRequest
+				// @brief Triggered when the cast service receives a hide request from a client
+				// @param applicationName: Registered application name
+				// @param applicationId: Application instance ID
+				virtual void OnApplicationHideRequest(const string& appName /* @text applicationName */ , const string& appID /* @text applicationId */ )  {};
+				// @text onApplicationCurrentStateRequest
+				// @brief 	Triggered when the cast service needs an update of the application state
+				// @param applicationName: Registered application name
+				// @param applicationId: Application instance ID
+				virtual void OnApplicationCurrentStateRequest(const string& appName /* @text applicationName */ , const string& appID /* @text applicationId */ )  {};
```

Comment:
@ramasat Do we really need current as it is implicit the state request will be for current only

Reply:
@apatel859 How about OnAppStateRequired?